tests/rules/entry-points.js

```diff
@@ -137,6 +149,19 @@ ruleTester.run('entry-points', rule, {
         '});'
       ].join('\n'),
       errors: [{ messageId: 'returnEntryPoint', data: { type: 'Restlet' }}]
+    },
+    {
+      code: [
+        '/**',
+        ' * @NScriptType ClientScript',
+        ' */',
+        'define([], function() {',
+        '  var exports = {};',
+        '  exports.notAnEntryPoint = x;',
+        '  return exports;',
+        '});'
+      ].join('\n'),
+      errors: [{ messageId: 'returnEntryPoint', data: { type: 'ClientScript' }}]
     }
   ]
 });
```

Comment:
Another check should be added to ensure that something like `notTheReturnObject.beforeLoad` won't count as a correct entry point either.